base/message_loop/message_pump_io_starboard.cc

```diff
@@ -181,6 +242,8 @@ void MessagePumpIOStarboard::Run(Delegate* delegate) {
   AutoReset<bool> auto_reset_keep_running(&keep_running_, true);
 
   for (;;) {
+    if (should_quit())
```

Comment:
Was this a bug previously or is it only needed now because of the socket changes?

Reply:
No, not a bug. Leftover from investigation of a bug in an earlier version of this change that had a different root cause. Removed.